app/routes/task_routes.py

```diff
@@ -0,0 +1,124 @@
+import requests
+import os
+from datetime import datetime
+from flask import Blueprint, jsonify, abort, make_response, request
+from dotenv import load_dotenv
+from app import db
+from app.models.task import Task
+from .routes_helpers import validate_model, find_date
+
+load_dotenv()
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+
+# CREATE ENDPOINT
+@task_bp.route("", methods=["POST"])
+def create_task():
+    req_body = request.get_json()
+
+    if "title" in req_body and "description" in req_body:
+        if "completed_at" in req_body:
+            date = find_date(req_body["completed_at" ])
+            new_task = Task.from_dict(req_body, date)
+        else:
+            new_task = Task.from_dict(req_body)
+    else:
+        abort(make_response({"details": "Invalid data"}, 400))
```

Comment:
Let's reshape this, so we can bring the main focus of the route/function outside indentation:
```python
    if "title" not in req_body or "description" not in req_body:
        abort(make_response({"details": "Invalid data"}, 400))

    new_task = Task.from_dict(req_body)
```
We should then check and assign the `completed_at` field inside the `from_dict` method!